blog/2024-08-14-headlamp/index.mdx

```diff
@@ -0,0 +1,86 @@
+authors: [team]
+tags: [opencost, headlamp, kubernetes, ui, integrations]
+---
+<head>
+    <link rel="canonical" href="https://headlamp.dev/blog/2024/08/08/opencost-plugin"/>
+</head>
+
+Reposted from the [Headlamp blog](https://headlamp.dev/blog/) post: [Streamlining Kubernetes Cost Management with the New OpenCost Plugin for Headlamp](https://headlamp.dev/blog/2024/08/08/opencost-plugin), by Santhosh Nagaraj
+
+![OpenCost + Headlamp](./img/splash.jpg)
+
+[OpenCost](https://www.opencost.io/) is one of CNCF's leading open-source projects that offers real-time native cost monitoring for Kubernetes environments, providing visibility to developers and companies on the costs linked to their cloud-native applications. With OpenCost, one can trace the costs of single workloads, namespaces, and even particular labels in your Kubernetes cluster. The continuous granular cost visibility that OpenCost provides help in effectual usage of resources and better budgeting.
+
+<!--truncate-->
+
+Given how useful OpenCost is, we created a [Headlamp](https://headlamp.dev/) plugin for OpenCost to help Headlamp users track their cluster resource costs. This integration aims to solve the visibility problem between cost and cluster management, enabling developers to make more informed decisions.
+
+:::note
+OpenCost must be installed with Prometheus metrics export, for the OpenCost plugin to display the data.
+:::
+
+Here is a quick video explaining the whole process:
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/J-jc0X8xk28?si=w_tmQW2cZgW9660k&cc_load_policy=1" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe>
+
+## Installing OpenCost in the cluster
+
+If you don’t have Prometheus and OpenCost installed in your cluster and you are using Headlamp as a desktop app, you can use the App Catalog to install them. This can be done by accessing the Apps from the sidebar, after you enter your cluster in Headlamp, look for the Prometheus app that is distributed by the prometheus-community, click install and follow the instructions.
+
+![Screenshot showing Prometheus Helm chart](./img/prometheus-helm-chart.png)
+
+Alternatively, you can install it by running the following commands.
+
+```bash
+helm repo add prometheus-community https://prometheus-community.github.io/helm-charts
+helm install prom prometheus-community/kube-prometheus-stack
+```
+
+Installing OpenCost:
+
+```yaml
+opencost:
+  prometheus:
+    internal:
+      enabled: true
+      namespaceName: <Prometheus-Namespace>
+      port: 9090
+      serviceName: <Prometheus-Service>
+```
+
+:::note
+Replace `<Prometheus-Namespace>` and `<Prometheus-Service>` with the namespace in which you have installed prometheus and prometheus service name.
+:::
+
+## Installing KubeCost plugin from the plugin Catalog
```

Comment:
Would you please update Kubecost to have a lowercase c?

Reply:
I think this was a typo. Should be OpenCost.